### PR TITLE
Added Google's model viewer within viewer_three.js

### PR DESCRIPTION
**Pull request by cconsta1**
## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ X] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [X ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ X] I have signed the CLA
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by cconsta1** [comment]
Hi! To test the changes please use the attached file containing 3D models :)
[models.zip](https://github.com/clowder-framework/clowder/files/9331470/models.zip)

**Review by cconsta1 — COMMENTED** [review]
I think I must have removed the console.log

File: public/javascripts/previewers/three_js/viewer_three.js [hunk 1]
```diff
@@ -42,247 +44,274 @@
...
-     const delta = clock.getDelta();
- 
-     if (mixer) mixer.update(delta);
- 
-     renderer.render(scene, camera);
- 
-     stats.update();
- }
\ No newline at end of file
+console.log("This is the 3D viewer");
```
[the remark this answers is not in the document]
Comment (cconsta1) [reply public/javascripts/previewers/three_js/viewer_three.js]: Hi @robkooper, I think I must have removed the `console.log`. Sorry, I was using it for testing the previewer and forgot to remove it. Please let me know if everything is ok now. Thanks!

[the remark this answers is not in the document]
Comment (robkooper) [reply public/javascripts/previewers/three_js/viewer_three.js]: no worries, i do the same and sometimes leave them. @lmarini will look at this on Monday, and I expect it to get included in the release later that day.

[the remark this answers is not in the document]
Comment (cconsta1) [reply public/javascripts/previewers/three_js/viewer_three.js]: @robkooper thank you very much, this sounds great!!

**Review by lmarini — CHANGES_REQUESTED** [review]
I noticed that when viewing .glb files the viewerport is much smaller than when viewing .fbx. Is that intentional?

Testing on firefox and chrome on mac. Otherwise it works great!

Screen Shot 2022-08-29 at 3 43 29 PM

Screen Shot 2022-08-29 at 3 43 17 PM

**Comment by cconsta1** [comment]
Hi Luigi,

Maybe this has something to do with the fact that I now used Google’s model viewer to load the GLB files instead of the three.js library (which I used before). But no it’s not intentional. I can check tomorrow. However, I did notice the viewport getting huge, which happens when I load 3D models with the developer tools open. I don’t know why that happens to be honest. To reproduce this, please load a file with the tools open and once it loads close the dev tools. 

Best regards,
Valentino

**Comment by lmarini** [comment]
> Hi Luigi, Maybe this has something to do with the fact that I now used Google’s model viewer to load the GLB files instead of the three.js library (which I used before). But no it’s not intentional. I can check tomorrow. However, I did notice the viewport getting huge, which happens when I load 3D models with the developer tools open. I don’t know why that happens to be honest. To reproduce this, please load a file with the tools open and once it loads close the dev tools. Best regards, Valentino
> […](#)
> Sent from my iPhone
> On Aug 29, 2022, at 11:48 PM, Luigi Marini ***@***.***> wrote: ﻿ @lmarini requested changes on this pull request. I noticed that when viewing .glb files the viewerport is much smaller than when viewing .fbx. Is that intentional? Testing on firefox and chrome on mac. — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you authored the thread.

@cconsta1 I tried with developer tools open on both firefox and chrome but the same thing happens.

**Comment by cconsta1** [comment]
Hi Luigi,

I think I know what the problem might be. The GLB file loads using the
element <model-viewer>. On my first try, the viewport was as small as yours
in the picture but I remedied that by adding a CSS entry:

model-viewer {
width: 640px;
height: 480px;
}

The CSS file (style.css) loads inside the viewer_three.js file. Somehow,
it's not working on your side. I can check tomorrow. I'm attaching what I'm
seeing on my side.

Regards,
Valentino

**Comment by lmarini** [comment]
Thank you, Valentino. So you have the same problem, but adding the model-viewer style at run time fixes it? Do you always have to add it after you refresh the page?

Thank you,
Luigi

From: Chrysovalantis Constantinou ***@***.***>
Date: Monday, August 29, 2022 at 5:17 PM
To: clowder-framework/clowder ***@***.***>
Cc: Marini, Luigi ***@***.***>, Mention ***@***.***>
Subject: Re: [clowder-framework/clowder] Added Google's model viewer within viewer_three.js (PR #372)
Hi Luigi,

I think I know what the problem might be. The GLB file loads using the
element <model-viewer>. On my first try, the viewport was as small as yours
in the picture but I remedied that by adding a CSS entry:

model-viewer {
width: 640px;
height: 480px;
}

The CSS file (style.css) loads inside the viewer_three.js file. Somehow,
it's not working on your side. I can check tomorrow. I'm attaching what I'm
seeing on my side.

Regards,
Valentino

**Comment by cconsta1** [comment]
Hi! I think I never answered this! What do you mean by loading the CSS style at run time?

**Review by lmarini — APPROVED** [review]

**Pull request merged**